variables.tf

```diff
@@ -70,6 +70,12 @@ variable "nat_gateway_enabled" {
   default     = true
 }
 
+variable "single_nat_gateway" {
```

Comment:
IMO, we should change this to `max_nat_gateways`. There's some code stink about calling it single, when single isn't the primary concern. It's the number of nat gateways to reduce cost. Operating in 5 AZs, it might be desirable to have 2 NAT gateways instead.